iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractBranchAdder.java

```diff
@@ -110,4 +121,24 @@ protected VoltageLevelExt checkAndGetVoltageLevel2() {
         return voltageLevel2;
     }
 
+    private String checkAndGetDefaultVoltageLevelId(String connectableBus) {
+        if (connectableBus != null) {
+            VoltageLevelExt defaultVoltageLevel = (VoltageLevelExt) getNetwork().getBusBreakerView()
+                    .getBus(connectableBus)
+                    .getVoltageLevel();
+            if (defaultVoltageLevel.getTopologyKind().equals(TopologyKind.BUS_BREAKER)) {
```

Comment:
I don't think this is the check that should be done: if the voltage level is `NODE_BREAKER`, it should fail later, not here necessarily.

The think to check is: if the bus `connectableBus` exists i.e. if `getNetwork().getBusBreakerView().getBus(connectableBus) != null` (it can not exist: return null, otherwise you will have a NPE)